apps/services/form-system/src/app/modules/forms/forms.service.ts

```diff
@@ -0,0 +1,359 @@
+    const formDto: FormDto = {
+      id: form.id,
+      organizationId: form.organizationId,
+      name: form.name,
+      urlName: form.urlName,
+      invalidationDate: form.invalidationDate,
+      created: form.created,
+      modified: form.modified,
+      isTranslated: form.isTranslated,
+      applicationDaysToRemove: form.applicationDaysToRemove,
+      derivedFrom: form.derivedFrom,
+      stopProgressOnValidatingScreen: form.stopProgressOnValidatingScreen,
+      completedMessage: form.completedMessage,
+      testimonyTypes: [],
+      applicants: [],
+      sections: [],
+      screens: [],
+      fields: [],
+    }
+
+    form.testimonyTypes?.map((testimonyType) => {
+      formDto.testimonyTypes?.push({
+        id: testimonyType.id,
+        name: testimonyType.name,
+        description: testimonyType.description,
+        type: testimonyType.type,
+      } as FormTestimonyTypeDto)
+    })
+
+    form.applicants?.map((applicant) => {
+      formDto.applicants?.push({
+        id: applicant.id,
+        applicantType: applicant.applicantType,
+        name: applicant.name,
+      } as FormApplicantDto)
+    })
+
+    form.sections.map((section) => {
+      formDto.sections?.push({
+        id: section.id,
+        name: section.name,
+        created: section.created,
+        modified: section.modified,
+        sectionType: section.sectionType,
+        displayOrder: section.displayOrder,
+        waitingText: section.waitingText,
+        isHidden: section.isHidden,
+        isCompleted: section.isCompleted,
+        callRuleset: section.callRuleset,
+      } as SectionDto)
+      section.screens?.map((screen) => {
+        formDto.screens?.push({
+          id: screen.id,
+          sectionId: section.id,
+          name: screen.name,
+          created: screen.created,
+          modified: screen.modified,
+          displayOrder: screen.displayOrder,
+          isHidden: screen.isHidden,
+          multiset: screen.multiset,
+        } as ScreenDto)
+        screen.fields?.map((field) => {
+          formDto.fields?.push({
+            id: field.id,
+            screenId: screen.id,
+            name: field.name,
+            created: field.created,
+            modified: field.modified,
+            displayOrder: field.displayOrder,
+            description: field.description,
+            isHidden: field.isHidden,
+            isPartOfMultiset: field.isPartOfMultiset,
+            fieldSettings:
+              this.fieldSettingsMapper.mapFieldTypeToFieldSettingsDto(
+                field.fieldSettings,
+                field.fieldType,
+              ),
+            fieldType: field.fieldType,
+          } as FieldDto)
+        })
+      })
+    })
+
+    return formDto
+  }
+
+  private async createFormTemplate(form: Form): Promise<void> {
+    await this.sectionModel.create({
+      formId: form.id,
+      sectionType: SectionTypes.PREMISES,
+      displayOrder: 0,
+      name: { is: 'Forsendur', en: 'Premises' },
```

Comment:
If all of these texts are supposed to be stored in code as strings it would be best to keep them together in translations files as objects and then reference them here

For example:

```javascript
// translations/form/create.ts

export const createFormTranslations: Record<string, LanguageType> = {
  premise: { is: 'Forsendur', en: Premises' },
}

// translations/form/index.ts
export * from './create'

// in code
import { createFormTranslations } from '../example/translations/form'

// ...
await this.sectionModel.create({
  name: createFormTranslations.premise
})
```

This way you will get typescript validation and you will keep changes to these files contained.